### PR TITLE
feat: add credentials management

### PR DESCRIPTION
**Pull request by vasilsatanasov**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/terraform-provider-vcf/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- `d/vcf_credentials` aded for extraction of credentials from the VCF instance
- `r/vcf_credentials_auto_rotate_policy` for managing of autorotate policy of the credentials
- `r/vcf_credentials_rotate` for explicit rotation of credentials
- `r/vcf_credentials_update` for explicit password change of VCF entities account
- added docs + examples

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

For bug fixes or features:

- [x] Tests have been completed.
- [x] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

**Review by dimitarproynov — COMMENTED** [review]
LGTM!

File: internal/provider/resource_credentials_auto_rotate_policy.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		data.Get("user_name").(string),
+	}
+
+	return credentials.HashFields(params)
+}
+
+func filterCredentialsByUsername(userName string, creds []*models.Credential) []*models.Credential {
+	result := make([]*models.Credential, 0)
+	for _, cred := range creds {
+		if *cred.Username == userName {
```
[the remark this answers is not in the document]
Comment (vasilsatanasov) [reply internal/provider/resource_credentials_auto_rotate_policy.go:138]: The SDK states 

// Username
// Required: true
Username *string `json:"username"`
but I will add an extra check

**Review by dimitarproynov — APPROVED** [review]

File: examples/resources/credentials/variables.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+variable "esxi_host_1" {
+  default = "10.10.01.01"
+  description = "Name of the first host in the setup"
+}
+
+variable "esxi_host_1_pass" {
+  default = "s0m3_p@$w0rd"
+  description = "The new password for the host"
+}
```
Comment (tenthirtyam) [inline examples/resources/credentials/variables.tf:9]:
```suggestion
variable "esxi_host_1_pass" {
  default = "s0m3_p@$w0rd"
  description = "The new password for the host"
  sensitive = true
}
```

**Review by tenthirtyam — APPROVED** [review]
LGTM!

Just a suggestion to:
- add `sensitve = true` in the example
- merge https://github.com/vmware/terraform-provider-vcf/pull/99 and then rebase this PR.

File: examples/resources/credentials/variables.tf (hunk 2 above)
Comment (vasilsatanasov) [reply examples/resources/credentials/variables.tf]: Done

**Review by dimitarproynov — APPROVED** [review]

**Pull request merged**